Dialog/TilesetEditDialog.cpp

```diff
@@ -1381,3 +1381,42 @@ void TilesetEditDialog::on_pushButton_changeBGTile8x8set_clicked()
         SetSelectedTile8x8(0, true);
     }
 }
+
+/// <summary>
+/// Change BG Tile8x8 set and the last palette by searching legal assorted graphic entries.
+/// </summary>
+void TilesetEditDialog::on_pushButton_CleanUpUnusedTile8x8_clicked()
+{
+    LevelComponents::Tileset *tmp_newTilesetPtr = tilesetEditParams->newTileset;
+    int existingTile8x8Num = tmp_newTilesetPtr->GetfgGFXlen() / 32;
+
+    // go through all the existing foreground Tile8x8s
+    for(int i = existingTile8x8Num; i > 0; i--)
```

Comment:
What about i == 0? should we delete that from the tileset?

Reply:
i == 0 is a blank tile8x8 should never be edited